src/webserver/WsBase.h

```diff
@@ -383,29 +366,57 @@ class WsSession : public WsBase,
 
         BOOST_LOG_TRIVIAL(debug)
             << __func__ << " received request from ip = " << *ip;
-        auto sendError = [&](auto&& msg) {
-            boost::json::object response;
-            response["error"] = std::move(msg);
-            std::string responseStr = boost::json::serialize(response);
 
-            BOOST_LOG_TRIVIAL(trace) << __func__ << " : " << responseStr;
+        auto sendError = [this, ip](
+                             auto error,
+                             const boost::json::value& id,
+                             boost::json::object& request) {
+            auto e = RPC::make_error(error);
+
+            if (!id.is_null())
+            {
+                e["id"] = id;
+            }
+            e["request"] = request;
 
+            auto responseStr = boost::json::serialize(e);
+            BOOST_LOG_TRIVIAL(trace) << __func__ << " : " << responseStr;
             dosGuard_.add(*ip, responseStr.size());
             send(std::move(responseStr));
         };
+
+        boost::json::value raw = [](std::string const&& msg) {
+            try
+            {
+                return boost::json::parse(msg);
+            }
+            catch (std::exception&)
+            {
+                return boost::json::value{nullptr};
+            }
+        }(std::move(msg));
+
+        if (!raw.is_object())
+            return sendError(
+                RPC::Error::rpcINVALID_PARAMS, nullptr, raw.as_object());
```

Comment:
This will throw, as raw is not an object. Just return a message that says something like "body is not json object"